pkg/wdk/primitives.go

```diff
@@ -120,6 +192,23 @@ type BEEF []byte
 // The TXID is given as a hex string followed by a period "." and then the output index is given as a decimal integer.
 type OutpointString string
 
+// Validate checks if the string is proper outpoint string and contains outpoint index after "."
+func (s OutpointString) Validate() error {
+	split := strings.Split(string(s), ".")
+
+	if len(split) != 2 {
+		return fmt.Errorf("txid as hexstring and numeric output index joined with '.'")
+	}
+
+	// check if after decimal point there is an outpoint index
+	_, err := strconv.Atoi(split[1])
```

Comment:
```suggestion
	_, err := to.Uint64FromString(split[1])
```
hopefully this will also ensure it's not negative number (If not, then we will need to fix need to fix it in `common` 😄